osu.Game/Overlays/TabbableOnlineOverlay.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Threading;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Online.API;
+
+namespace osu.Game.Overlays
+{
+    public abstract class TabbableOnlineOverlay<THeader, TEnum> : OnlineOverlay<THeader>
```

Comment:
All the logic minus tab control really don't make sense being in this class specifically, instead of at the base `OnlineOverlay` class instead.

But either way, I still don't get what the exact goal and use of this PR is, currently it looks to be doing incomplete abstractions on overlays seeing that API and display logic are in this class, and I'm not entirely sure if that's something fine to be as-is. When instead all online overlays should have that logic as well, and even if that happens, it'll make this PR look to be doing many things at once.

If the goal of this PR is for having a shared drawable layout (`ScrollFlow -> FillFlow -> (Header, Content)`) on all overlays (which would probably be useful for fixing the loading state covering headers pretty easily), then that can be included in this PR solely.

Reply:
I presume the usage of `OnlineOverlay` is going to come up in a future PR? @EVAST9919 can you provide some more information regarding this?

Reply:
The point of this pr was organizing the overlays (since some of them are literally copy-pasted) and moving header creation from the ctor parameter to a property. It will hopefully help to make loading logic implementation much easier.
Regarding `TabbableOnlineOverlay` - I disagree. The logic here is quite different from one-page overlays (such as `BeatmapListng`) and having it in the base class is excessive, because it will never be used for those.
But sure: one may say that the current logic is totally okay and I'm making things even more complicated. Either way explanation above is my personal opinion and I may be wrong.

Reply:
Tabbing logic should definitely *not* be in the base. How would that work on non-tabbed overlays - by a property or something? Runs completely counter to the entire idea of composition/inheritance/whatever have you.

Given there's more than one implementation of this pattern I don't see extracting this out as necessarily bad.

Reply:
Well, the main concern I have with it is that it looks to be abstracted in a pretty odd way. 

You wouldn't expect a method that loads and displays certain content to the overlay cancellable by API state change to be in a `TabbableOnlineOverlay` class, it's not logic specific for tabbing.

Perhaps the abstraction could've been done better, in a way that doesn't involve a manual `LoadDisplay` method call, something like an `UpdateTabDisplay` for updating display and `CreateTabDisplay` for creating tab display.

Reply:
I don't have the slightest idea what your problem is exactly, and how your two generic-named methods are supposed to fix things or even fit into this structure. If you care enough, write some actual code and we'll talk.

Reply:
Yeah I don't know then, if you don't see any problem with the way this is abstracted then I guess it's not something to care much about.

Also have attempted writing code but seems that `RankingsOverlay` requires some rewriting to fit into the structure properly, well then I guess this is fine for now.